base/compiler/ssair/ir.jl

```diff
@@ -1420,18 +1421,23 @@ function iterate(it::CompactPeekIterator, (idx, aidx, bidx)::NTuple{3, Int}=(it.
     return (compact.ir.stmts[idx][:inst], (idx + 1, aidx, bidx))
 end
 
-# This Union{Nothing, Pair{Pair{Int,Int},Any}} cannot be stack allocated, so we inline it
-@inline function iterate(compact::IncrementalCompact, st::Tuple{Int, Int}=(compact.idx, 1))
-    st = iterate_compact(compact, st)
+struct IncrementalCompactState end
```

Comment:
Is it worth being this explicit over e.g. just using `nothing` as the state?

Reply:
Ah yeah, of course, I guess I was confused by `iterate(...) = nothing` being indicative of the end-of-iteration, but here we return `something, nothing`.